File: pY/edgeaiinference.py
# edge_ai_inference.py
# Simulates inference of a lightweight model at the edge

class EdgeModel:
    def predict(self, x):
        # Simple rule-based model: classify as 'active' if input > 0.5
        return "active" if x > 0.5 else "inactive"

model = EdgeModel()
data = [0.2, 0.6, 0.8, 0.3]

for d in data:
    print(f"Input: {d:.2f} => Result: {model.predict(d)}")

File: R/edgeaiinference.R
# edge_ai_inference.R
predict_edge <- function(x) {
  if (x > 0.5) "active" else "inactive"
}

data <- c(0.2, 0.6, 0.8, 0.3)
for (d in data) {
  cat("Input:", d, "=> Result:", predict_edge(d), "\n")
}
